src/datasets/iterable_dataset.py

```diff
@@ -2211,6 +2250,7 @@ def map(
         remove_columns: Optional[Union[str, List[str]]] = None,
         features: Optional[Features] = None,
         fn_kwargs: Optional[dict] = None,
+        format_outputs: bool = True,
```

Comment:
is this useful ?

Reply:
was useful for optimising performance in my case - 

idea is it allows the user to determine the formatting of the examples returned by map (within the mapped function) rather than via self.formatting, which determines the formatting of the inputs to the map function

if map returns lists / arbitrary types then re-applying the formatter to the outputs can be expensive

However, haven't thought about how to handle self.formatting for downstream dataset transformations...might require something better than this kwarg

Reply:
You can still do `ds = ds.map().with_format(None)` and no formatting will be applied to the output

Reply:
perfect - will remove the kwarg